### PR TITLE
Parsed downselection of sidesets from other sidesets

### PR DESCRIPTION
**Pull request by GiudGiud**
Replaces #24363

Uses the ParsedGenerateSideset

### CODE REVIEW

**Comment by GiudGiud** [comment]
There's a rider commit on nodeset generation. If we dont have that the nodeset gets generated from the sideset after automartically. This MG has a line setting the nodeset name, so it seems like it should generate the nodeset.

Alternatively I can remove the line setting the nodeset name at the end, or add a parameter to select if the nodeset should be generated or not.

i think it s all pretty equivalent.

File: framework/src/meshgenerators/ParsedGenerateSideset.C [hunk 1]
```diff
@@ -191,13 +200,29 @@ ParsedGenerateSideset::generate()
...
       std::unique_ptr<Elem> curr_side = elem->side_ptr(side);
       _func_params[0] = curr_side->vertex_average()(0);
       _func_params[1] = curr_side->vertex_average()(1);
       _func_params[2] = curr_side->vertex_average()(2);
       if (evaluate(_func_F))
+      {
         boundary_info.add_side(elem, side, boundary_ids[0]);
+        // Create nodeset by default
+        for (auto & node : curr_side->node_ref_range())
+          boundary_info.add_node(&node, boundary_ids[0]);
```
Comment (lindsayad) [inline framework/src/meshgenerators/ParsedGenerateSideset.C:224]:
```suggestion
```
I don't like this very much. The class is suffixed with `Sideset`. We already build node lists from side lists by default anyway after mesh generation is all done. You can remove the line in this class where we set the name of the nodeset for more consistency

File: test/tests/meshgenerators/parsed_generate_sideset/tests [hunk 2]
```diff
@@ -23,7 +23,20 @@
...
+  []
+
+  [limit_to_one_bid]
+    type = 'Exodiff'
+    input = 'parsed_generate_sideset_boundary_limited.i'
+    cli_args = '--mesh-only'
+    exodiff = 'parsed_generate_sideset_boundary_limited_in.e'
+    mesh_mode = 'REPLICATED'
+    recover = false
+
```
Comment (lindsayad) [inline test/tests/meshgenerators/parsed_generate_sideset/tests:35]:
```suggestion
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Failures in Sodium are unrelated

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**